enhancements/update/update-status-api.md

```diff
@@ -0,0 +1,398 @@
+
+The `status` subresource content is the primary value of the enhancement. The status and health information about the update are expressed through the `status` subresource.
+
+Introduce a new `update.openshift.io` group for `UpdateStatus`. There is no suitable existing OpenShift API group for where `UpdateStatus` would fit. The `ClusterVersion` CRD, which is used to convey some information about the update process, is in the `config.openshift.io` group. `UpdateStatus` provides no configuration capabilities, so `config.openshift.io` is a poor fit. The new group is well-suited to contain some of the other update-related APIs needed for further incoming features like maintenance windows or update pre-checks.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+The enhancement allows adoption in HCP as an extension. The API and controller will be extended to handle the resources that represent the control plane and worker pools (`HostedCluster` and `NodePool`, respectively) and surface update progress through status insights for these resources.
+
+The Update Status Controller will run as a part of the hosted control plane. Unlike in Standalone, where it is a CVO operand, USC will be managed by the Control Plane Controller in HCP, where the CVO does not have access to management cluster API and cannot manage resources there. USC needs to access the management cluster API layer to manage the status-reporting API (which targets cluster administrators tasked with updating clusters) and needs to report insights about both the hosted cluster and management cluster resources. It will also need access to the hosted cluster API layer to report on resources present there.
+
+Because `UpdateStatus` API is cluster-scoped, it is not suitable for HCP use, and we will need to introduce a namespaced HCP-specific flavor, such as `HostedClusterUpdateStatus`. HCP-specific API variant will also need to be able to refer to resources in two API surfaces (management and hosted clusters), while the `UpdateStatus` API can rely on all resources being in the same cluster and use simple resource references.
+
+The updates in HCP are not initiated by the [`hcp` client][hcp-client] binary but by [directly modifying `HostedCluster` and `NodePool` resources][updates-in-hcp] using the `oc` client. Hence, at the moment, there is no apparent suitable place to put the update status UX (a client that would consume a `HostedClusterUpdateStatus` instance and present its content). The best place for that is the [`hcp` client][hcp-client], if necessary. It is expected that managed OCP product layers using HCP (ROSA) would consume `HostedClusterUpdateStatus` to improve their update experience, but that work is outside the scope of this enhancement.
```

Comment:
fwiw if there was a use case the hcp could push the namespaced UpdateStatus into the guest cluster as a cluster scoped UpdateStatus. I'd expect though there's no need for that and this is consumed by managed services which them choose how to present to the service consumer

Reply:
>  I'd expect though there's no need for that and this is consumed by managed services which them choose how to present to the service consumer

+1